tools/api-markdown-documenter/src/LintApiModel.ts

```diff
@@ -0,0 +1,302 @@
+	 */
+	readonly kind: TError;
+}
+
+/**
+ * An error resulting from a reference tag with an invalid target.
+ */
+export interface ReferenceError {
+	/**
+	 * Name of the item that included a reference to an invalid target.
+	 */
+	readonly sourceItem: string;
+
+	/**
+	 * The string provided as the reference in a reference tag.
+	 */
+	readonly referenceTarget: string;
+
+	/**
+	 * The name of the package that the {@link ReferenceError.sourceItem} belongs to.
+	 */
+	readonly packageName: string;
+}
+
+/**
+ * An error resulting from an `{@inheritDoc}` tag with an invalid target.
+ */
+// eslint-disable-next-line unicorn/prevent-abbreviations
+export interface InheritDocReferenceError
+	extends LinterErrorBase<LinterErrorKind.InheritDocReferenceError>,
+		ReferenceError {
+	/**
+	 * {@inheritDoc LinterErrorBase.kind}
+	 */
+	readonly kind: LinterErrorKind.InheritDocReferenceError;
+}
+
+/**
+ * An error resulting from a `{@link}` tag with an invalid target.
+ */
+export interface LinkReferenceError
+	extends LinterErrorBase<LinterErrorKind.LinkReferenceError>,
+		ReferenceError {
+	/**
+	 * {@inheritDoc LinterErrorBase.kind}
+	 */
+	readonly kind: LinterErrorKind.LinkReferenceError;
+}
+
+/**
+ * {@link LinterResult} base type.
+ */
+export interface LinterResultBase<TSuccess extends boolean> {
+	/**
+	 * Result "success" status.
+	 */
+	readonly success: TSuccess;
+}
+
+/**
+ * A successful linter result. No issues were found.
+ */
+export interface LinterSuccessResult extends LinterResultBase<true> {
+	/**
+	 * {@inheritDoc LinterResultBase.status}
+	 */
+	readonly success: true;
+}
+
+/**
+ * Success result singleton.
+ */
+const successResult: LinterSuccessResult = { success: true };
+
+/**
+ * A linter failure result. One or more issues were found.
+ */
+export interface LinterFailureResult extends LinterResultBase<false> {
+	/**
+	 * {@inheritDoc LinterResultBase.status}
+	 */
+	readonly success: false;
+
+	/**
+	 * `{@inheritDoc}` reference errors found in the API model.
+	 */
+	readonly inheritDocReferenceErrors: readonly InheritDocReferenceError[];
+
+	/**
+	 * `{@link}` reference errors found in the API model.
+	 */
+	readonly linkReferenceErrors: readonly LinkReferenceError[];
+}
+
+/**
+ * Result of {@link lintApiModel}.
+ * @remarks Includes a {@link LinterResult.success} indicating whether or not the check was a "success" (no issues found) or "failure" (issues found).
+ */
+export type LinterResult = LinterSuccessResult | LinterFailureResult;
+
+/**
+ * Validates the given API model.
+ *
+ * @returns A "Result" object indicating whether or not the check was a "success" (no issues found) or "failure" (issues found).
+ * In the case of a "failure", all issues found are included.
+ *
+ * @throws
+ * If the specified {@link LoadModelOptions.modelDirectoryPath} doesn't exist, or if no `.api.json` files are found directly under it.
+ */
+export async function lintApiModel(options: LintApiModelOptions): Promise<LinterResult> {
```

Comment:
For reviewers: this is the crux of the PR. A function for loading an API Model (all package `api.json` files) and running validation checks on it (for now, limited to validating reference targets of `{@inheritDoc}` comments, but more checks will be added in subsequent PRs).